src/test/java/org/springframework/cloud/scheduler/spi/cloudfoundry/CloudFoundryAppSchedulerTests.java

```diff
@@ -0,0 +1,369 @@
+import org.springframework.cloud.deployer.spi.core.AppDefinition;
+import org.springframework.cloud.scheduler.spi.core.ScheduleRequest;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.core.io.Resource;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+import static org.springframework.cloud.scheduler.spi.core.SchedulerPropertyKeys.CRON_EXPRESSION;
+
+/**
+ * Test the core features of the Spring Cloud Scheduler implementation.
+ *
+ * @author Glenn Renfro
+ */
+public class CloudFoundryAppSchedulerTests {
+
+	public static final String DEFAULT_CRON_EXPRESSION = "* * * * *";
+
+	@Mock(answer = Answers.RETURNS_SMART_NULLS)
+	private Applications applications;
+
+	@Mock(answer = Answers.RETURNS_SMART_NULLS)
+	private CloudFoundryOperations operations;
+
+	@Mock(answer = Answers.RETURNS_SMART_NULLS)
+	private Spaces spaces;
+
+	@Mock(answer = Answers.RETURNS_SMART_NULLS)
+	private ApplicationsV3 applicationsV3;
+
+	@Mock(answer = Answers.RETURNS_SMART_NULLS)
+	private CloudFoundryClient cloudFoundryClient;
+
+	@Mock(answer = Answers.RETURNS_SMART_NULLS)
+	private Tasks tasks;
+
+	@Mock(answer = Answers.RETURNS_SMART_NULLS)
+	private CloudFoundry2630AndLaterTaskLauncher taskLauncher;
+
+	private CloudFoundryAppScheduler cloudFoundryAppScheduler;
+
+	private SchedulerClient client;
+
+	private CloudFoundryConnectionProperties properties = new CloudFoundryConnectionProperties();
+
+
+	@Before
+	public void setUp(){
+		MockitoAnnotations.initMocks(this);
+		given(this.cloudFoundryClient.applicationsV3()).willReturn(this.applicationsV3);
+		given(this.cloudFoundryClient.tasks()).willReturn(this.tasks);
+		given(this.spaces.list()).willReturn(getTestSpaces());
+
+		this.properties.setSpace("test-space");
+
+		given(this.operations.applications()).willReturn(this.applications);
+		given(this.operations.spaces()).willReturn(this.spaces);
+
+		this.client = new TestSchedulerClient();
+
+		this.cloudFoundryAppScheduler = new CloudFoundryAppScheduler(this.client, this.operations,
+				this.properties, taskLauncher);
+	}
+
+	@Test(expected = IllegalArgumentException.class)
+	public void testEmptySchedulerProperties() {
+		Resource resource = new FileSystemResource("src/test/resources/demo-0.0.1-SNAPSHOT.jar");
+		AppDefinition definition = new AppDefinition("bar", null);
+		ScheduleRequest request = new ScheduleRequest(definition, null, null, null, resource);
+		this.cloudFoundryAppScheduler.schedule(request);
+	}
+
+	@Test
+	public void testCreateNoCommandLineArgs() {
+		Resource resource = new FileSystemResource("src/test/resources/demo-0.0.1-SNAPSHOT.jar");
+
+		mockAppResultsInAppList();
+		AppDefinition definition = new AppDefinition("test-application-1", null);
+		ScheduleRequest request = new ScheduleRequest(definition, getDefaultScheduleProperties(), null, "test-schedule", resource);
```

Comment:
Small nit, `ScheduleRequest` could be more aligned with the constructor argument order of `AppDeploymentRequest`, 

```
	public AppDeploymentRequest(AppDefinition definition, Resource resource,
			Map<String, String> deploymentProperties, List<String> commandlineArguments) {
```

and `ScheduleRequest` ctor is

```
	public ScheduleRequest(AppDefinition definition,  Map<String, String> schedulerProperties,
			Map<String, String> deploymentProperties, List<String> commandlineArguments,
			String scheduleName, Resource resource) {
```

The resource is last where as before it was second in line.   What about

```
	public ScheduleRequest(AppDefinition definition,   Resource resource, 
                        Map<String, String> schedulerProperties, String scheduleName, 
			Map<String, String> deploymentProperties, List<String> commandlineArguments,) {
```